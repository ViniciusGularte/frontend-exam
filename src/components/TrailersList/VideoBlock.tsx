import React from "react";
import {
  ContainerVideoBlock,
  ContainerVideoBlockImage,
  VideoLabel,
} from "./style";

function VideoBlock({
  title,
  thumbnail,
}: {
  title: string;
  thumbnail: string;
}) {
  console.log(title);
  return (
    <ContainerVideoBlock background={thumbnail}>
      <ContainerVideoBlockImage background={thumbnail}>
        <VideoLabel>{title}</VideoLabel>
      </ContainerVideoBlockImage>
    </ContainerVideoBlock>
  );
}
export default VideoBlock;
