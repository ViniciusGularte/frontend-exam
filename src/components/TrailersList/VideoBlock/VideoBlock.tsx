import React from "react";
import {
  ContainerVideoBlock,
  ContainerVideoBlockImage,
  VideoLabel,
} from "./style";

function VideoBlock({
  title,
  idVideo,
  thumbnail,
  tooglePopup,
}: {
  title: string;
  idVideo: string;
  thumbnail: string;
  tooglePopup: (x: string) => void;
}) {
  const tooglePopupRequest = () => {
    tooglePopup(idVideo);
  };
  return (
    <ContainerVideoBlock onClick={tooglePopupRequest}>
      <ContainerVideoBlockImage background={thumbnail}>
        <VideoLabel>{title}</VideoLabel>
      </ContainerVideoBlockImage>
    </ContainerVideoBlock>
  );
}
export default VideoBlock;
