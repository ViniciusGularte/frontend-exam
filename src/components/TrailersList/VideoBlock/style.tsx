import styled from "styled-components";
import { devicesMax } from "../../../assets/global_style";
const ContainerVideoBlock = styled.div`
  display: flex;
  flex: 1 0 40%;
  flex-direction: row;
  height: 232px;
  margin-left: 40px;
  margin-right: 40px;
  margin-top: 80px;
  border: 1px solid ${(props) => props.theme.secondary};
  cursor: pointer;
  transition: transform 0.2s;
  &:hover {
    transform: scale(1.1);
  }

  @media ${devicesMax.mobileL} {
    flex: 1 0 80%;
    margin: 16px;
  }
`;
const ContainerVideoBlockImage = styled.div`
  border: 1px solid #131313;

  flex-direction: row;
  display: flex;
  flex: 1;
  background-image: url(${(props) => props.background});
  background-repeat: no-repeat;
  background-size: cover;
  object-fit: contain;
  background-position: 0px;
  height: 230px;
`;

const VideoLabel = styled.div`
  flex: 1;
  align-self: flex-end;
  text-align: center;
  color: ${(props) => props.theme.primary};
  background-color: black;
  padding: 8px;
  opacity: 0.7;
`;
export { ContainerVideoBlock, ContainerVideoBlockImage, VideoLabel };
