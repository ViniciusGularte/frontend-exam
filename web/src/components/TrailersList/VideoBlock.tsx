import React from "react";
import { ContainerVideoBlock } from "./style";
function VideoBlock() {
  return <ContainerVideoBlock>hello</ContainerVideoBlock>;
}
export default VideoBlock;
