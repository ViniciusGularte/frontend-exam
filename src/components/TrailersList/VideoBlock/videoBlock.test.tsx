import Enzyme, { mount } from "enzyme";
import ReactSeventeenAdapter from "@wojtekmaj/enzyme-adapter-react-17";
import VideoBlock from "./VideoBlock";

Enzyme.configure({ adapter: new ReactSeventeenAdapter() });
const mockCallBack = jest.fn();

const infoVideo = {
  key: 0,
  idVideo: "543535",
  tooglePopup: mockCallBack,
  thumbnail: "witcher",
  title: "the witcher",
};
describe("VideoBlock", () => {
  it("mount correctly", () => {
    const wrapper = mount(
      <VideoBlock
        key={infoVideo.key}
        idVideo={infoVideo.idVideo}
        tooglePopup={infoVideo.tooglePopup}
        thumbnail={infoVideo.thumbnail}
        title={infoVideo.title}
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("div")).toHaveLength(2);
    expect(wrapper.find("span").text()).toEqual("the witcher");
  });
});
