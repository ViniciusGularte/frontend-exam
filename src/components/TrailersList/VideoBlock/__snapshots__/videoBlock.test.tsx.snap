// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoBlock mount correctly 1`] = `ReactWrapper {}`;
